ocaml/src/remotes.ml

```diff
@@ -166,21 +166,28 @@ module Pool = struct
          Kinetic_client.make_client buffer_pool ~conn_info kinetic_id
 
     let use_osd t ~(osd_id:int32) f =
-      let pool =
-        try Hashtbl.find t.pools osd_id
+      let get_pool () =
+        try Hashtbl.find t.pools osd_id |> Lwt.return
         with Not_found ->
-          let p =
-            Lwt_pool2.create
-              t.pool_size
-              ~check:(fun _ exn ->
-                  (* TODO some exns shouldn't invalidate the connection *)
-                  false)
-              ~factory:(fun () -> t.get_osd_kind osd_id >>= factory t.tls_config t.buffer_pool)
-              ~cleanup:(fun (_, closer) -> closer ())
-          in
-          Hashtbl.add t.pools osd_id p;
-          p
+          begin
+            t.get_osd_kind osd_id >>= fun kind ->
```

Comment:
`kind` contains the connection info, which could in fact change.
It looks like in osd_access we however do some effort to invalidate a pool if the connection info is deemed to be incorrect, so probably this optimisation is not breaking anything.